tests/functional/restart/54-incompatible-db.t

```diff
@@ -14,32 +14,36 @@
 #
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#-------------------------------------------------------------------------------
-# Compat: Cylc<8
-# Test the upgrade of the old *retrying states to the new xtrigger based
-# retry mechanism.
+#------------------------------------------------------------------------------
+# Test that restarting a Cylc 7 suite does not work due to database
+# incompatibility, and that suitable error message is given
+
```

Comment:
Can this be done with an integration test instead?

Reply:
Kinda, but not so easy, this is fine.